docs/introduction/en/storage.md

```diff
@@ -0,0 +1,53 @@
+
+![](/img/storage-formula.png)
+
+When account α becomes the owner of a stored entry (whether it is created or modified), α should immediately lock 1/16 CFX for the entry. If α has enough balance, then the required deposit will be automatically locked, otherwise if α does not have enough balance, the operation will fail and α cannot create or modify the entry.
+
+When a storage item is deleted from the world state, the corresponding 1/16 CFX deposit will be unlocked and returned to the item owner's balance. If the ownership of a storage item changes, the 1/16 CFX deposit of the old owner is unlocked, and the new owner must lock 1/16 CFX as the deposit at the same time. It is worth mentioning that the deposit refund is "quietly" added to the balance, and there is no transfer transaction available for inquiry.
+
+### Solidity memory management mechanism
+
+According to the [Description](https://solidity-cn.readthedocs.io/zh/develop/introduction-to-smart-contracts.html#index-10) of the Solidity document on its memory management and Conflux [Storage Deposit Mechanism](https://mp.weixin.qq.com/s/KKU3XiJqr_Uz5RdlUyQmtA), we can find that the contract storage needs to be maintained with key and value. In general: the length of key is 256bits, and the length of value is also 256bits. The storage space of a smart contract is organized according to the following table, where {0,1}<sup>256</sup> represents a 256-bit string (there are only two values ​​of 0 or 1 in the bit string), **each key/value pair can be understood as an entry**:
+
+Entry | Key/Address({0,1}<sup>256</sup>)   | value({0,1}<sup>256</sup>)  |
+|  :----:  |  :----:  | :----:  |
+1| 0...00000  | 0 |
+2| 0...00001  | 1 |
+3| 0...00002 | 2 |
+...| ... | ... |
+2<sup>256</sup>| f...fffff| 0 |
+
+Since 256bits=32bytes, the corresponding length of two 256bits is: 32+32=64 (Bytes)
+
+Summary of common variables and their corresponding entries in Solidity:
+
+|   Variable   |                            Length                             |                      Definition Method                       |
+| :------: | :---------------------------------------------------------: | :-------------------------------------------------: |
+| Ordinary variables |                   1 ordinary variable corresponds to 1 entry                    |                uint public count=0;                 |
+| mapping  |              Each key of mapping corresponds to 1 entry              |      mapping(address => uint) public balances;      |
+|  array   | Each element of the array corresponds to 1 item, and the length of the array arr.length is an additional item | uint[5] fixedArr = [1,2,3,4,5]; string productname; |
```

Comment:
>  Each element of the array corresponds to 1 item, and the length of the array arr.length is an additional item

I think array elements can be "packed tightly" (e.g. with `uint128`). But we do not need to go into so many details in this doc.

Reply:
Yeah, i have remove this part